test/tools/ossfuzz/protomutators/YulProtoMutator.h

```diff
@@ -0,0 +1,130 @@
+#pragma once
+
+#include <test/tools/ossfuzz/yulProto.pb.h>
+
+#include <libsolutil/Common.h>
+
+#include <src/libfuzzer/libfuzzer_macro.h>
+
+#include <random>
+
+namespace solidity::yul::test::yul_fuzzer
+{
+
+using ProtobufMessage = google::protobuf::Message;
+using ProtobufDesc = google::protobuf::Descriptor;
+
+template <typename Proto>
+using LPMPostProcessor = protobuf_mutator::libfuzzer::PostProcessorRegistration<Proto>;
+
+template <typename Proto>
+using FuzzMutatorCallback = std::function<void(Proto*, unsigned)>;
+
+template <typename P>
+struct YulProtoCBRegistration
+{
+	YulProtoCBRegistration(FuzzMutatorCallback<P> const& _callback)
+	{
+		LPMPostProcessor<P> reg = {_callback};
+	}
+};
+
+class MutationInfo: public ScopeGuard
+{
+public:
+	MutationInfo(ProtobufMessage const* _message, std::string const& _info);
+
+	static void writeLine(std::string const& _str)
+	{
+		std::cout << _str << std::endl;
+	}
+	void exitInfo();
+
+	ProtobufMessage const* m_protobufMsg;
+};
+
+struct YulRandomNumGenerator
+{
+	using RandomEngine = std::minstd_rand;
+
+	explicit YulRandomNumGenerator(unsigned _seed): m_random(RandomEngine(_seed)) {}
+
+	unsigned operator()()
+	{
+		return m_random();
+	}
+
+	RandomEngine m_random;
+};
+
+template <typename Proto>
+using CustomFuzzMutator = std::function<void(Proto*, YulRandomNumGenerator& _rand)>;
+
+struct YulProtoMutator
+{
+	enum class PrintChanges { No, Yes };
+
+	template <typename T>
+	static void functionWrapper(
+		CustomFuzzMutator<T> const& _callback,
+		T* _message,
+		unsigned _seed,
+		unsigned _period,
+		std::string const& _info,
+		PrintChanges _printChanges = PrintChanges::No
+	);
+
+	/// Return an integer literal of the given value
```

Comment:
This comment seems redundant since there is a more descriptive `@returns` tag below